app/helpers/ems_physical_infra_helper/textual_summary.rb

```diff
@@ -54,7 +54,12 @@ def textual_port
   end
 
   def textual_physical_switches
-    textual_link(@record.physical_switches)
```

Comment:
most likely, adding `:as => PhysicalSwitch` would have the same effect

More to the point, the new code does not use `role_allows?`, so it can't stay as is.

Reply:
Sorry for that, this was an old solution that I put before fixing the base model for PhysicalSwitch. I put back the textual link call, which is working with new solution.